test/mocks/event/mocks.h

```diff
@@ -174,6 +174,38 @@ class MockTimer : public Timer {
   Event::TimerCb callback_;
 };
 
+class MockRangeTimer : public RangeTimer {
+public:
+  MockRangeTimer();
+  ~MockRangeTimer() override;
+
+  void invokeCallback() {
+    EXPECT_TRUE(enabled_);
+    enabled_ = false;
+    if (scope_ == nullptr) {
+      callback_();
+      return;
+    }
+    ScopeTrackerScopeState scope(scope_, *dispatcher_);
```

Comment:
You need to accept dispatcher as a constructor argument, otherwise you'll run problems with this line.